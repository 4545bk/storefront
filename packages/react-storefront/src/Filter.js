/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import withStyles from '@material-ui/core/styles/withStyles'
import { get } from 'lodash';
import { observer, inject } from 'mobx-react';
import ExpandableSection from './ExpandableSection'
import Checkbox from '@material-ui/core/Checkbox'
import FormGroup from '@material-ui/core/FormGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import PropTypes from 'prop-types'
import { types } from "mobx-state-tree"
import { Hbox } from './Box'

/**
 * UI for filtering an instance of SearchResultModelBase.  This component can be used on its own, or you can use
 * FilterButton to automatically display this component in a drawer that slides up from the bottom of the viewport.
 */
export const styles = theme => ({
  root: {
    height: '50vh',
    overflowY: 'auto',
    paddingBottom: '64px'
  },
  matches: {
    marginLeft: '5px',
    display: 'inline'
  },
  footer: {
    backgroundColor: theme.palette.secondary.main,
    padding: '12px 20px',
    color: 'white',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    justifyContent: 'space-between',
    alignItems: 'center',
    transform: 'translate3d(0,0,0)'
  },
  itemsFound: {
    color: theme.palette.secondary.contrastText
  }
});

@withStyles(styles, { name: 'RSFFilter' })
@inject('router')
@observer
export default class Filter extends Component {

  static propTypes = {
    /**
     * An instance of `SearchResultModelBase`
     */
    model: PropTypes.object,

    /**
     * CSS classes
     */
    classes: PropTypes.object,

    /**
     * A function to call when the user clicks the button to view updated results.  The default behavior can be
     * canceled by calling `preventDefault` on the passed in event.  The event is passed as the only argument.
     */
    onViewResultsClick: PropTypes.func,

    /**
     * The query string parameter that should be updated when filters are changed.  The value will be an array
     * of codes for each selected facet.  Defaults to "filters"
     */
    queryParam: PropTypes.string
  }

  static defaultProps = {
    onViewResultsClick: Function.prototype,
    queryParam: 'filters'
  }

  state = {
    expanded: {}
  }

  render() {
    const { model, classes } = this.props

    return (
      <div className={classes.root}>
       { get(model, 'facetGroups', []).map((facetGroup, i) => this.renderFacetGroup(facetGroup, i)) }
       { model.filtersChanged && (
        <Hbox className={classes.footer} split>
          <Typography variant="subheading" className={classes.itemsFound}>{model.filters.length || 'No'} filter{model.filters.length === 1 ? '' : 's'} selected</Typography>
          <Button variant="raised" size="large" color="default" onClick={this.onViewResultsClick}>View Results</Button>
        </Hbox>
       )}
      </div>
    )
  }

  renderFacetGroup(group, key) {
    const { app, model, classes } = this.props
    const { expanded } = this.state
    const selection = []

    const formGroup = (
      <FormGroup key={key}>
        { group.facets.map((facet, i) => {
          let checked = false

          if (get(model, 'filters', []).indexOf(facet.code) !== -1) {
            selection.push(facet)
            checked = true
          }

          return (
            <FormControlLabel
              key={i}
              label={
                <span>
                  <span>{facet.name}</span>
                  <Typography variable="caption" className={classes.matches} component="span">({facet.matches})</Typography>
                </span>
              }
              control={
                <Checkbox
                  checked={checked}
                  color="primary"
                  onChange={this.onToggleFilter.bind(this, facet)}
                />
              }
            />
          )
        })}
      </FormGroup>
    )

    let caption = null

    if (selection.length === 1) {
      caption = selection[0].name
    } else if (selection.length > 0) {
      caption = `${selection.length} selected`
    }

    return (
      <ExpandableSection 
        key={key}
        title={group.name} 
        caption={caption}
        expanded={expanded[group.name]}
        onChange={(e, expanded) => this.setState({ expanded: { ...expanded, [group.name]: expanded }})}
      >
        {formGroup}
      </ExpandableSection>
    )
  }

  onToggleFilter = (facet) => {
    this.props.model.toggleFilter(facet)
  }

  onViewResultsClick = (e) => {
    const { router, model } = this.props

    this.props.onViewResultsClick(e)

    if (!e.isDefaultPrevented()) {
      this.props.model.setFiltersChanged(false)
      router.applySearch({
        [this.props.queryParam]: model.filters.toJSON()
      }) 
    }
  }

}

