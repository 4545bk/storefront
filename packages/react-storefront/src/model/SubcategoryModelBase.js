/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import { types } from "mobx-state-tree"
import SearchResultsModelBase from './SearchResultsModelBase'

const SubcategoryModelBase = types.compose(SearchResultsModelBase, 
  types.model("SubcategoryModelBase", {
    id: types.identifier(types.string),
    url: types.maybe(types.string),
    name: types.maybe(types.string),
    description: types.maybe(types.string)
  })
  .views(self => ({
    shouldApplyPatchOnPop(patch) {
      return patch.page === 'Subcategory'
    }
  }))
)

export default SubcategoryModelBase
