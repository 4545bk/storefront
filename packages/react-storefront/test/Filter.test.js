/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { create } from 'react-test-renderer'
import SubcategoryModelBase from '../src/model/SubcategoryModelBase'
import Filter from '../src/Filter'
import Provider from './TestProvider'

function createModel(filters) {
  return SubcategoryModelBase.create({
    id: '1',
    filters,
    facetGroups: [{
      name: 'Price',
      facets: [{
        code: '0-100',
        name: 'Up to $100',
        matches: 20
      }, {
        code: '100-200',
        name: '$100 - $200',
        matches: 10
      }]
    }, {
      name: 'Color',
      facets: [{
        code: 'red',
        name: 'Red',
        matches: 2
      }, {
        code: 'sunburst',
        name: 'Sunburst',
        matches: 10
      }]
    }]
  })
}

describe('Filter', () => {

  it('renders with no filters selected', () => {
    const component = (
      <Provider router={{}} app={{}}>
        <Filter model={createModel([])}/>
      </Provider>
    )

    expect(create(component).toJSON()).toMatchSnapshot()
  })

  it('renders with filters selected', () => {
    const component = (
      <Provider router={{}} app={{}}>
        <Filter model={createModel(['0-100', 'sunburst'])}/>
      </Provider>
    )

    expect(create(component).toJSON()).toMatchSnapshot()
  })

})