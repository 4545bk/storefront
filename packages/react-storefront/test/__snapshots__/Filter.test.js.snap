// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders with filters selected 1`] = `
<div
  className="RSFFilter-root-139"
>
  <div
    className="MuiPaper-root-161 MuiPaper-elevation1-164 MuiExpansionPanel-root-158 RSFExpandableSection-root-146 RSFExpandableSection-margins-147 RSFExpandableSection-expandedPanel-155"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-194 MuiExpansionPanelSummary-root-188"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-192 RSFExpandableSection-summary-152"
      >
        <h6
          className="MuiTypography-root-197 MuiTypography-subtitle1-215"
        >
          Price
        </h6>
        <span
          className="MuiTypography-root-197 MuiTypography-caption-207 RSFExpandableSection-caption-148"
        >
          Up to $100
        </span>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiExpansionPanelSummary-expandIcon-193 RSFExpandableSection-summaryIconWrap-154"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-237"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-238 RSFExpandableSection-expandIcon-156"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-270"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-247"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-249"
      >
        <div
          className="MuiCollapse-wrapperInner-250"
        >
          <div
            className="MuiExpansionPanelDetails-root-251 RSFExpandableSection-details-151"
          >
            <div
              className="MuiFormGroup-root-252"
            >
              <label
                className="MuiFormControlLabel-root-254"
              >
                <span
                  className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiPrivateSwitchBase-root-266 MuiCheckbox-root-260 MuiCheckbox-colorPrimary-264 MuiPrivateSwitchBase-checked-267 MuiCheckbox-checked-261"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-237"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-238"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                    <input
                      checked={true}
                      className="MuiPrivateSwitchBase-input-269"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-270"
                  />
                </span>
                <span
                  className="MuiTypography-root-197 MuiTypography-body2-205 MuiFormControlLabel-label-259"
                >
                  <div
                    className="RSFFilter-groupLabel-145"
                  >
                    <span>
                      Up to $100
                    </span>
                    <span
                      className="MuiTypography-root-197 MuiTypography-caption-207 RSFFilter-matches-140"
                    >
                      (
                      20
                      )
                    </span>
                  </div>
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root-254"
              >
                <span
                  className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiPrivateSwitchBase-root-266 MuiCheckbox-root-260 MuiCheckbox-colorPrimary-264"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-237"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-238"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-269"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-270"
                  />
                </span>
                <span
                  className="MuiTypography-root-197 MuiTypography-body2-205 MuiFormControlLabel-label-259"
                >
                  <div
                    className="RSFFilter-groupLabel-145"
                  >
                    <span>
                      $100 - $200
                    </span>
                    <span
                      className="MuiTypography-root-197 MuiTypography-caption-207 RSFFilter-matches-140"
                    >
                      (
                      10
                      )
                    </span>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-161 MuiPaper-elevation1-164 MuiExpansionPanel-root-158 RSFExpandableSection-root-146 RSFExpandableSection-margins-147 RSFExpandableSection-expandedPanel-155"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-194 MuiExpansionPanelSummary-root-188"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-192 RSFExpandableSection-summary-152"
      >
        <h6
          className="MuiTypography-root-197 MuiTypography-subtitle1-215"
        >
          Color
        </h6>
        <span
          className="MuiTypography-root-197 MuiTypography-caption-207 RSFExpandableSection-caption-148"
        >
          Sunburst
        </span>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiExpansionPanelSummary-expandIcon-193 RSFExpandableSection-summaryIconWrap-154"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-237"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-238 RSFExpandableSection-expandIcon-156"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-270"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-247"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-249"
      >
        <div
          className="MuiCollapse-wrapperInner-250"
        >
          <div
            className="MuiExpansionPanelDetails-root-251 RSFExpandableSection-details-151"
          >
            <div
              className="MuiFormGroup-root-252"
            >
              <label
                className="MuiFormControlLabel-root-254"
              >
                <span
                  className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiPrivateSwitchBase-root-266 MuiCheckbox-root-260 MuiCheckbox-colorPrimary-264"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-237"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-238"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-269"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-270"
                  />
                </span>
                <span
                  className="MuiTypography-root-197 MuiTypography-body2-205 MuiFormControlLabel-label-259"
                >
                  <div
                    className="RSFFilter-groupLabel-145"
                  >
                    <span>
                      Red
                    </span>
                    <span
                      className="MuiTypography-root-197 MuiTypography-caption-207 RSFFilter-matches-140"
                    >
                      (
                      2
                      )
                    </span>
                  </div>
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root-254"
              >
                <span
                  className="MuiButtonBase-root-194 MuiIconButton-root-232 MuiPrivateSwitchBase-root-266 MuiCheckbox-root-260 MuiCheckbox-colorPrimary-264 MuiPrivateSwitchBase-checked-267 MuiCheckbox-checked-261"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-237"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-238"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                    <input
                      checked={true}
                      className="MuiPrivateSwitchBase-input-269"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-270"
                  />
                </span>
                <span
                  className="MuiTypography-root-197 MuiTypography-body2-205 MuiFormControlLabel-label-259"
                >
                  <div
                    className="RSFFilter-groupLabel-145"
                  >
                    <span>
                      Sunburst
                    </span>
                    <span
                      className="MuiTypography-root-197 MuiTypography-caption-207 RSFFilter-matches-140"
                    >
                      (
                      10
                      )
                    </span>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filter renders with no filters selected 1`] = `
<div
  className="RSFFilter-root-1"
>
  <div
    className="MuiPaper-root-23 MuiPaper-elevation1-26 MuiExpansionPanel-root-20 RSFExpandableSection-root-8 RSFExpandableSection-margins-9 RSFExpandableSection-expandedPanel-17"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-56 MuiExpansionPanelSummary-root-50"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-54 RSFExpandableSection-summary-14"
      >
        <h6
          className="MuiTypography-root-59 MuiTypography-subtitle1-77"
        >
          Price
        </h6>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiExpansionPanelSummary-expandIcon-55 RSFExpandableSection-summaryIconWrap-16"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-99"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-100 RSFExpandableSection-expandIcon-18"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-132"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-109"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-111"
      >
        <div
          className="MuiCollapse-wrapperInner-112"
        >
          <div
            className="MuiExpansionPanelDetails-root-113 RSFExpandableSection-details-13"
          >
            <div
              className="MuiFormGroup-root-114"
            >
              <label
                className="MuiFormControlLabel-root-116"
              >
                <span
                  className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiPrivateSwitchBase-root-128 MuiCheckbox-root-122 MuiCheckbox-colorPrimary-126"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-99"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-100"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-131"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-132"
                  />
                </span>
                <span
                  className="MuiTypography-root-59 MuiTypography-body2-67 MuiFormControlLabel-label-121"
                >
                  <div
                    className="RSFFilter-groupLabel-7"
                  >
                    <span>
                      Up to $100
                    </span>
                    <span
                      className="MuiTypography-root-59 MuiTypography-caption-69 RSFFilter-matches-2"
                    >
                      (
                      20
                      )
                    </span>
                  </div>
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root-116"
              >
                <span
                  className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiPrivateSwitchBase-root-128 MuiCheckbox-root-122 MuiCheckbox-colorPrimary-126"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-99"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-100"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-131"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-132"
                  />
                </span>
                <span
                  className="MuiTypography-root-59 MuiTypography-body2-67 MuiFormControlLabel-label-121"
                >
                  <div
                    className="RSFFilter-groupLabel-7"
                  >
                    <span>
                      $100 - $200
                    </span>
                    <span
                      className="MuiTypography-root-59 MuiTypography-caption-69 RSFFilter-matches-2"
                    >
                      (
                      10
                      )
                    </span>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-23 MuiPaper-elevation1-26 MuiExpansionPanel-root-20 RSFExpandableSection-root-8 RSFExpandableSection-margins-9 RSFExpandableSection-expandedPanel-17"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-56 MuiExpansionPanelSummary-root-50"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-54 RSFExpandableSection-summary-14"
      >
        <h6
          className="MuiTypography-root-59 MuiTypography-subtitle1-77"
        >
          Color
        </h6>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiExpansionPanelSummary-expandIcon-55 RSFExpandableSection-summaryIconWrap-16"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-99"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-100 RSFExpandableSection-expandIcon-18"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-132"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-109"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-111"
      >
        <div
          className="MuiCollapse-wrapperInner-112"
        >
          <div
            className="MuiExpansionPanelDetails-root-113 RSFExpandableSection-details-13"
          >
            <div
              className="MuiFormGroup-root-114"
            >
              <label
                className="MuiFormControlLabel-root-116"
              >
                <span
                  className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiPrivateSwitchBase-root-128 MuiCheckbox-root-122 MuiCheckbox-colorPrimary-126"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-99"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-100"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-131"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-132"
                  />
                </span>
                <span
                  className="MuiTypography-root-59 MuiTypography-body2-67 MuiFormControlLabel-label-121"
                >
                  <div
                    className="RSFFilter-groupLabel-7"
                  >
                    <span>
                      Red
                    </span>
                    <span
                      className="MuiTypography-root-59 MuiTypography-caption-69 RSFFilter-matches-2"
                    >
                      (
                      2
                      )
                    </span>
                  </div>
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root-116"
              >
                <span
                  className="MuiButtonBase-root-56 MuiIconButton-root-94 MuiPrivateSwitchBase-root-128 MuiCheckbox-root-122 MuiCheckbox-colorPrimary-126"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-99"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-100"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-131"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-132"
                  />
                </span>
                <span
                  className="MuiTypography-root-59 MuiTypography-body2-67 MuiFormControlLabel-label-121"
                >
                  <div
                    className="RSFFilter-groupLabel-7"
                  >
                    <span>
                      Sunburst
                    </span>
                    <span
                      className="MuiTypography-root-59 MuiTypography-caption-69 RSFFilter-matches-2"
                    >
                      (
                      10
                      )
                    </span>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
